test/e2e/utils.go

```diff
@@ -0,0 +1,134 @@
+	return waitInterface(namespace, nodeName, name, false)
+}
+
+func writePodsLogs(namespace string, writer io.Writer) error {
+	if framework.Global.LocalOperator {
+		return nil
+	}
+	podLogOpts := corev1.PodLogOptions{}
+	podList := &corev1.PodList{}
+	err := framework.Global.Client.List(context.TODO(), &dynclient.ListOptions{}, podList)
+	Expect(err).ToNot(HaveOccurred())
+	podsClientset := framework.Global.KubeClient.CoreV1().Pods(namespace)
+
+	for _, pod := range podList.Items {
+		req := podsClientset.GetLogs(pod.Name, &podLogOpts)
+		podLogs, err := req.Stream()
+		if err != nil {
+			io.WriteString(writer, fmt.Sprintf("error in opening stream: %v\n", err))
+			continue
+		}
+		defer podLogs.Close()
+		_, err = io.Copy(writer, podLogs)
+		if err != nil {
+			io.WriteString(writer, fmt.Sprintf("error in copy information from podLogs to buf: %v\n", err))
+			continue
+		}
+
+	}
+	return nil
+}
+
+func prepare(t *testing.T) (*framework.TestCtx, string) {
+	By("Initialize cluster resources")
+	cleanupRetryInterval := time.Second * 1
+	cleanupTimeout := time.Second * 5
+	ctx := framework.NewTestCtx(t)
+	err := ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	Expect(err).ToNot(HaveOccurred())
+
+	// get namespace
+	namespace, err := ctx.GetNamespace()
+	Expect(err).ToNot(HaveOccurred())
+
+	// wait for memcached-operator to be ready
+	err = e2eutil.WaitForOperatorDeployment(t, framework.Global.KubeClient, namespace, "kubernetes-nmstate", 1, time.Second*5, time.Second*30)
+	Expect(err).ToNot(HaveOccurred())
+	return ctx, namespace
+}
+
+func readStateFromFile(file string, namespace string) nmstatev1.NodeNetworkState {
+	manifest, err := ioutil.ReadFile(file)
+	Expect(err).ToNot(HaveOccurred())
+
+	state := nmstatev1.NodeNetworkState{}
+	err = yaml.Unmarshal(manifest, &state)
+	Expect(err).ToNot(HaveOccurred())
+	state.ObjectMeta.Namespace = namespace
+	return state
+}
+
+func createStateFromFile(file string, namespace string, cleanupOptions *framework.CleanupOptions) {
+	state := readStateFromFile(file, namespace)
+	err := framework.Global.Client.Create(context.TODO(), &state, cleanupOptions)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func updateStateSpecFromFile(file string, key types.NamespacedName) {
+	state := nmstatev1.NodeNetworkState{}
+	stateFromManifest := readStateFromFile(file, key.Namespace)
+	err := framework.Global.Client.Get(context.TODO(), key, &state)
+	Expect(err).ToNot(HaveOccurred())
+	state.Spec = stateFromManifest.Spec
+	err = framework.Global.Client.Update(context.TODO(), &state)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func nodeNetworkState(key types.NamespacedName) nmstatev1.NodeNetworkState {
+	state := nmstatev1.NodeNetworkState{}
+	Eventually(func() error {
+		return framework.Global.Client.Get(context.TODO(), key, &state)
+	}).ShouldNot(HaveOccurred())
```

Comment:
the default timeout is a second, please be explicit setting it.

Reply:
Done

Reply:
How is being implicit more clear? The default timeout is a second with polling interval 10 ms. 1 second might be fine, but is it really necessary to flood Kubernetes API server with 10ms polling?

Reply:
Missunderstood it, agree with 10ms being too much, going to change it to 1 second interval and 5 seconds timeout.